docs/content/FormControl.mdx

```diff
@@ -0,0 +1,414 @@
+  </FormControl>
+</Box>
+```
+
+### With validation
+
+<Note>Validation messages are not used for an individual checkbox or radio form control.</Note>
+
+```javascript live noinline
+const ValidationExample = () => {
+  const [value, setValue] = React.useState('mona lisa')
+  const [validationResult, setValidationResult] = React.useState()
+  const doesValueContainSpaces = inputValue => /\s/g.test(inputValue)
+  const handleInputChange = e => {
+    setValue(e.currentTarget.value)
+  }
+
+  React.useEffect(() => {
+    if (doesValueContainSpaces(value)) {
+      setValidationResult('noSpaces')
+    } else if (value) {
+      setValidationResult('validName')
+    }
+  }, [value])
+
+  return (
+    <FormControl>
+      <FormControl.Label>GitHub handle</FormControl.Label>
+      <TextInput block value={value} onChange={handleInputChange} />
+      {validationResult === 'noSpaces' && (
+        <FormControl.Validation>GitHub handles cannot contain spaces</FormControl.Validation>
+      )}
+      {validationResult === 'validName' && <FormControl.Validation>Valid name</FormControl.Validation>}
+      <FormControl.Caption>With or without "@". For example "monalisa" or "@monalisa"</FormControl.Caption>
+    </FormControl>
+  )
+}
+
+render(ValidationExample)
+```
+
+### With a leading visual
+
+<Note>Only a checkbox or radio form control may have a leading visual.</Note>
+
+```jsx live
+<>
+  <FormControl variant="choice">
+    <FormControl.Label>Option one</FormControl.Label>
+    <FormControl.LeadingVisual>
+      <MarkGithubIcon />
+    </FormControl.LeadingVisual>
+    <Checkbox />
+  </FormControl>
+
+  <FormControl variant="choice">
+    <FormControl.Label>Option two</FormControl.Label>
+    <FormControl.LeadingVisual>
+      <MarkGithubIcon />
+    </FormControl.LeadingVisual>
+    <Checkbox />
+    <FormControl.Caption>This one has a caption</FormControl.Caption>
+  </FormControl>
+</>
+```
+
+## Props
+
+### FormControl
+
+The container that handles the layout and passes the relevant IDs and ARIA attributes it's children.
+
+<PropsTable>
+  <PropsTableRow
+    name="children"
+    type="FormControl.Label | FormControl.Caption | FormControl.Validation | Autocomplete | TextInput | TextInputWithTokens | Select"
+    required
+  />
+  <PropsTableRow
+    name="variant"
+    defaultValue="'stack'"
+    type="'stack' | 'choice'"
+    description="Changes the layout of the form control"
+  />
+  <PropsTableRow
+    name="disabled"
+    type="boolean"
+    defaultValue="false"
+    description="Whether the field is ready for user input"
+  />
+  <PropsTableRow
+    name="id"
+    type="string"
+    defaultValue="a generated string"
+    description="The unique identifier for this field. Used to associate the label, validation text, and caption text"
+  />
+  <PropsTableRow
+    name="required"
+    type="boolean"
+    defaultValue="false"
+    description="Whether the field is ready for user input"
+  />
+</PropsTable>
```

Comment:
Do you intentionally omit the `sx` prop from the `FormControl.*` components?

Reply:
Yes